src/arrayMethodJoin.js

```diff
@@ -4,9 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        result += separator;
+      } else {
+        result += this[i];
+        result += separator;
+      }
+    }
+
+    if (separator === '') {
```

Comment:
`  if (separator === '') {
      return (result);
    }

    if (separator === null) {
      return (result.slice(0, -4));
    }

    return result.slice(0, -separator.toString().length);`
    
    You can make this part short if check 'i' only at last iteration.
    
    But it's work, and let the mentor decide =)